### PR TITLE
add _candidates_ja/LucDuponcheel_1.md

### LINKED ISSUE #343: 和訳：Luc Duponcheelさんによる「Program Description Based Programming」
ref #335

### PR DESCRIPTION
**Pull request by richardimaoka**
Closes #343

```
name: Luc Duponcheel
title: "Program Description Based Programming"
length: 40
audience: Advanced
language: English
twitter: @LucDup
github: https://github.com/LucDupAtGitHub
icon: https://avatars0.githubusercontent.com/u/77954?s=400&u=7f67fc0da57a4ef97a65349c4f7cd39f40557812&v=4
organization: ImagineJ
tags:
  - Functional Programming, Best Practices
suggestions:
  - After attending the talk the attendee will appreciate the flexibility of separating program descriptions from their executable interpretations (active, reactive, ...).
---
In his Turing award winning lecture, John Backus presented FP, a programming system promoting a point-free functional programming style based upon Functions, Composition, Construction and Condition. FP closely corresponds to Arrows as introduced by John Hughes. Point-free programming has, traditionally, been considered to be rather cumbersome. The good news is that Scala comes to the rescue by spicing this style with a DSL flavor that makes point-free program descriptions almost look like native Scala programs. The talk goes both into the details of the design of a library and it's usage. Among others it goes into details on how implicit functions can greatly reduce boilerplate code.
```

### CODE REVIEW

File: _candidates_ja/LucDuponcheel_1.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+icon: https://avatars0.githubusercontent.com/u/77954?s=400&u=7f67fc0da57a4ef97a65349c4f7cd39f40557812&v=4
+organization: ImagineJ
+tags:
+  - Functional Programming
+  - Best Practices
+suggestions:
+  - このトークを聞けば参加者は、高レベルな記述部分(description)と低レベル(でアクティブもしくはリアクティブ)な実行部分(interpretation)とを分離することによって実現されるプログラムの柔軟性を理解することが出来ます
+---
+チューリング章を受賞したJohn Backusは自身の講義のなかで、関数型プログラミング、すなわち関数、関数合成、ConstructionおよびConditionに基づく
+ポイントフリーな関数型プログラミングスタイルを提唱しました。
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: In his Turing award winning lecture, John Backus presented FP, a programming system promoting a point-free functional programming style based upon Functions, Composition, Construction and Condition.

File: _candidates_ja/LucDuponcheel_1.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+関数型プログラミングはJohn Hughesが示したように、数学的な射影と深い関わりがあります。
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: FP closely corresponds to Arrows as introduced by John Hughes.

File: _candidates_ja/LucDuponcheel_1.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+伝統的にポイントフリープログラミングは実現するには面倒なものだと考えられていましたが、
+Scalaを使えばDSLによってポイントフリープログラミングをほとんどネイティブなScalaプログラムと変わらない見た目に保つことが出来ます。
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: Point-free programming has, traditionally, been considered to be rather cumbersome. The good news is that Scala comes to the rescue by spicing this style with a DSL flavor that makes point-free program descriptions almost look like native Scala programs.

File: _candidates_ja/LucDuponcheel_1.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+このトークはライブラリのデザインにおける詳細および使用方法について話し、
+そしてimplicit関数がいかにしてボイラープレートなコードを削減するかを紹介します。
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: The talk goes both into the details of the design of a library and it's usage. Among others it goes into details on how implicit functions can greatly reduce boilerplate code.

File: _candidates_ja/LucDuponcheel_1.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+language: English
+twitter: LucDup
+github: LucDupAtGitHub
+icon: https://avatars0.githubusercontent.com/u/77954?s=400&u=7f67fc0da57a4ef97a65349c4f7cd39f40557812&v=4
+organization: ImagineJ
+tags:
+  - Functional Programming
+  - Best Practices
+suggestions:
+  - このトークを聞けば参加者は、高レベルな記述部分(description)と低レベル(でアクティブもしくはリアクティブ)な実行部分(interpretation)とを分離することによって実現されるプログラムの柔軟性を理解することが出来ます
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: After attending the talk the attendee will appreciate the flexibility of separating program descriptions from their executable interpretations (active, reactive, ...).

File: _candidates_ja/LucDuponcheel_1.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
+---
+name: Luc Duponcheel
+title: "記述的な方法による関数型プログラミング"
```
Comment (richardimaoka) [inline _candidates_ja/LucDuponcheel_1.md]: "Program Description Based Programming"

File: _candidates_ja/LucDuponcheel_1.md (hunk 2 above)
Comment (eed3si9n) [reply _candidates_ja/LucDuponcheel_1.md]: Arrow の訳は猫番では「射」としているので、「射 (Arrow) と深い関わりがあります」とかどうでしょうか。

http://eed3si9n.com/herding-cats/ja/Arrow.html

File: _candidates_ja/LucDuponcheel_1.md (hunk 1 above)
Comment (eed3si9n) [reply _candidates_ja/LucDuponcheel_1.md]: typo: チューリング章
チューリング賞

File: _candidates_ja/LucDuponcheel_1.md (hunk 6 above)
Comment (eed3si9n) [reply _candidates_ja/LucDuponcheel_1.md]: 訳案: 記述的プログラミング

タイトルには fp という言葉は出てこないので、なるべくシンプルにしました。

**Comment by richardimaoka** [comment]
ありがとうございます。レビュー反映しました @eed3si9n

**Pull request merged**